### PR TITLE
Eclipse/prashanth/fix/refresh taskview on close open project

### PR DESCRIPTION
**Pull request by RaoPrashanth**
Supports auto-refreshing of a closing/opening operation on projects in workspace

### CODE REVIEW

File: org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/ProjectClosedEvent.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2016 the original author or authors.
```
Comment (donat) [inline org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/ProjectClosedEvent.java]: 2018

File: org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/ProjectOpenedEvent.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2016 the original author or authors.
```
Comment (donat) [inline org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/ProjectOpenedEvent.java]: 2018

File: org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/ProjectChangeListener.java [hunk 3]
```diff
@@ -78,6 +79,12 @@ private boolean doVisitDelta(IResourceDelta delta) throws Exception {
                 } else if (fromPath != null) {
                     CorePlugin.listenerRegistry().dispatch(new ProjectMovedEvent(project, fromPath.lastSegment()));
                 }
+            } else if (delta.getFlags() == IResourceDelta.OPEN) {
+            	if (project.isOpen()) {
+            		CorePlugin.listenerRegistry().dispatch(new ProjectOpenedEvent(project));
+            	} else if (!project.isOpen()) {
```
Comment (donat) [inline org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/ProjectChangeListener.java]: Why not just a simple `else`?

File: org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/task/WorkspaceProjectsChangeListener.java [hunk 4]
```diff
@@ -38,7 +39,8 @@ public WorkspaceProjectsChangeListener(TaskView taskView) {
 
     @Override
     public void onEvent(Event event) {
-        if (event instanceof ProjectCreatedEvent || event instanceof ProjectDeletedEvent || event instanceof GradleNatureAddedEvent) {
+        if (event instanceof ProjectCreatedEvent || event instanceof ProjectDeletedEvent || event instanceof GradleNatureAddedEvent
```
Comment (donat) [inline org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/task/WorkspaceProjectsChangeListener.java]: Please add test coverage for the changes to the `TaskViewContentTest` class.

Comment (RaoPrashanth) [reply org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/task/WorkspaceProjectsChangeListener.java]: Thanks for the review. I have made other 2 changes. But I have added a test case as well. But when I try to run the test I am getting following error.
image

Comment (donat) [reply org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/task/WorkspaceProjectsChangeListener.java]: Judging from the icons of the `.groovy` files it seems that your Eclipse doesn't have Groovy tooling installed. Did you set up your [development environment](https://github.com/eclipse/buildship/blob/master/docs/development/Setup.md) via Oomph?

Comment (RaoPrashanth) [reply org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/task/WorkspaceProjectsChangeListener.java]: Oh, I will install the Groovy plugins into Eclipse. 
Yes, I followed the same steps mentioned in the link.

But I will it a try once again. Give me some time. 
Thanks for helping me!!

**Comment by donat** [comment]
@RaoPrashanth Sorry for the delay, I was drawn into other activities. I'll have another look soon, and then I'll get back to you.

**Comment by donat** [comment]
Btw, the IP-validation (⬆️) failed for this PR, can you please fix it. You could probably rebase your PR to the latest master and squash the changes into a single commit, which has the [required signature](https://wiki.eclipse.org/Development_Resources/Contributing_via_Git#via_GitHub).

**Comment by RaoPrashanth** [comment]
I will reopen a new pull request with proper commit with sign-off.

**Comment by donat** [comment]
@RaoPrashanth any update on the new PR?

**Pull request closed without merging**